esdk-performance-testing/benchmarks/go/benchmark/benchmark_tests.go

```diff
@@ -0,0 +1,453 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package benchmark
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"runtime"
+	"runtime/metrics"
+	"sort"
+	"sync"
+	"time"
+
+	esdktypes "github.com/aws/aws-encryption-sdk/releases/go/encryption-sdk/awscryptographyencryptionsdksmithygeneratedtypes"
+	"github.com/schollz/progressbar/v3"
+)
+
+// === Helper Functions ===
+
+// runEncryptDecryptCycle performs a single encrypt-decrypt cycle and measures performance
+func (b *ESDKBenchmark) runEncryptDecryptCycle(data []byte) (float64, float64, error) {
+	ctx := context.Background()
+
+	encryptionContext := map[string]string{
+		"purpose": "performance-test",
+		"size":    fmt.Sprintf("%d", len(data)),
+	}
+
+	encryptInput := esdktypes.EncryptInput{
+		Plaintext:         data,
+		Keyring:           b.Keyring,
+		EncryptionContext: encryptionContext,
+	}
+
+	// Encrypt
+	encryptStart := time.Now()
+	encryptOutput, err := b.EsdkClient.Encrypt(ctx, encryptInput)
+	if err != nil {
+		return 0, 0, fmt.Errorf("encryption failed: %w", err)
+	}
+	encryptDuration := time.Since(encryptStart).Seconds() * 1000
+
+	// Decrypt
+	decryptInput := esdktypes.DecryptInput{
+		Ciphertext: encryptOutput.Ciphertext,
+		Keyring:    b.Keyring,
+	}
+
+	decryptStart := time.Now()
+	decryptOutput, err := b.EsdkClient.Decrypt(ctx, decryptInput)
+	if err != nil {
+		return 0, 0, fmt.Errorf("decryption failed: %w", err)
+	}
+	decryptDuration := time.Since(decryptStart).Seconds() * 1000
+
+	// Verify integrity
+	if !bytes.Equal(data, decryptOutput.Plaintext) {
+		return 0, 0, fmt.Errorf("data integrity check failed")
+	}
+
+	return encryptDuration, decryptDuration, nil
+}
+
+// shouldRunTestType checks if a test type should be run based on quick config
+func (b *ESDKBenchmark) shouldRunTestType(testType string) bool {
+	if b.Config.QuickConfig == nil || len(b.Config.QuickConfig.TestTypes) == 0 {
+		return true
+	}
+
+	for _, allowedType := range b.Config.QuickConfig.TestTypes {
```

Comment:
nit: If we change TestTypes to a map of strings and boolean instead of list, shouldRunTestType might get faster. example:
```
func (b *ESDKBenchmark) shouldRunTestType(testType string) bool {
    if b.allowedTestTypes == nil {
        return true
    }
    return b.allowedTestTypes[testType]
}
```

Reply:
True. But practically it would be negligible - I don't think we'd ever see 100+ test types for this category (and string comparison is also not an expensive operation).